packages/server/src/api/controllers/table/index.ts

```diff
@@ -40,6 +40,20 @@ function pickApi({ tableId, table }: { tableId?: string; table?: Table }) {
   return internal
 }
 
+function checkDefaultFields(table: Table) {
+  for (const [key, field] of Object.entries(table.schema)) {
+    if (!("default" in field) || field.default == null) {
+      continue
+    }
+    if (field.constraints?.presence) {
```

Comment:
We should use `packages/shared-core/src/helpers/schema.ts, isRequired`. This might be a complex object, and this function encapsulates the logic

Reply:
Just got a little bitten by this because the type that `isRequired` accepts is `FieldConstraints | undefined`. Because all of the fields in `FieldConstraints` are optional, any object can be successfully passed in. So I initially tried `isRequired(field)` instead of `isRequired(field.constraints)` and nothing shouted at me, but all the tests failed.

I'm not sure what the answer is, but it's a bit of a sharp edge as it currently stands.